public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -397,22 +401,25 @@ function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
     const result: AnnotationQueryKind = {
       kind: 'AnnotationQuery',
       spec: {
+        builtIn: Boolean(layer.state.query.builtIn),
         name: layer.state.query.name,
         datasource: layer.state.query.datasource || getDefaultDataSourceRef(),
-        query: {
-          kind: getAnnotationQueryKind(layer.state.query),
-          spec: omit(layer.state.query, 'datasource'),
-        },
         enable: Boolean(layer.state.isEnabled),
         hide: Boolean(layer.state.isHidden),
         filter: layer.state.query.filter ?? defaultAnnotationPanelFilter(),
         iconColor: layer.state.query.iconColor,
-        builtIn:
-          layer.state.query.builtIn === undefined
-            ? Boolean(layer.state.query.builtIn)
-            : defaultAnnotationQuerySpec().builtIn,
       },
     };
+
+    // Check if DataQueryKind exists
+    const queryKind = getAnnotationQueryKind(layer.state.query);
+    if (layer.state.query.query?.kind === queryKind) {
```

Comment:
We only persist query if `DataQueryKind` exists in AnnotationsDataLayer. When transforming to scene, we pass the whole `AnnotationQueryKind` to `DashboardAnnotationsDataLayer.query`, which is defined here:
https://github.com/grafana/grafana/blob/a59e7e14b1fe68524010542eae0a8161e7f50b65/packages/grafana-data/src/types/annotations.ts#L12-L23

Since `AnnotationQueryKind` has query field, the only reason why this works is because a definition above allows `[key: string]: any;` and that's why when transforming back to schema we need to check if `AnnotationsDataLayerState` which has query field at the root level, also has the query field we passed with `AnnotationQueryKind`.